pkg/endpoint/policy.go

```diff
@@ -210,7 +210,17 @@ func (e *Endpoint) determineAllowLocalhost(desiredPolicyKeys map[policymap.Polic
 			Identity:         identityPkg.ReservedIdentityHost.Uint32(),
 			TrafficDirection: policymap.Ingress.Uint8(),
 		}
+
+		// Remove eventual existing ingress L4 localhost restrictions
+		for k, key := range desiredPolicyKeys {
+			if key.Identity == identityPkg.ReservedIdentityHost.Uint32() &&
```

Comment:
oh, also, the key object is the key of the map. `k` is the object with all the stuff in it, I'm afraid.